Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -859,58 +877,101 @@ def partitionByScript(
         # script-specific one.
         if scripts & DFLT_SCRIPTS:
             scripts = COMMON_SCRIPTS_SET
-        for script in scripts:
-            side1Scripts.setdefault(script, set()).add(glyph)
+        resolvedScripts[glyph] = scripts
+        scripts = sorted(scripts)
+        directions = [script_direction(script) for script in scripts]
+        for direction in directions:
```

Comment:
same as https://github.com/googlefonts/ufo2ft/pull/811/files#r1461703599
```suggestion
        for direction in (script_direction(script) for script in scripts):
```